apps/server/lib/lexical/server/project/progress/util.ex

```diff
@@ -0,0 +1,15 @@
+defmodule Lexical.Server.Project.Progress.Util do
+  alias Lexical.Server.Project.Progress
+
+  import Lexical.RemoteControl.Api.Messages
+
+  def with_progress(project, label, func) do
+    progress_pid = Progress.name(project)
+
+    send(progress_pid, project_progress(label: label <> ".begin"))
+    result = func.()
```

Comment:
this should probably try/rescue to ensure the end progress is sent.

Reply:
This function is only called in the "start project node" function. If it cannot be launched, it still will send an end event. A try-catch block may not be necessary for now.

Reply:
Another point is the issue of user experience. I'm not sure if it's necessarily a good thing to be unable to send an end message when encountering an exception, as this would force users to check logs for issues. Personally, I like doing it this way.

Reply:
won't that just cause a message to be left open? That's not great; I'd rather tell them that it failed.